lib/operations/collection_ops.js

```diff
@@ -401,7 +399,10 @@ function distinct(coll, key, query, options, callback) {
 
   options = Object.assign({}, options);
   // Ensure we have the right read preference inheritance
-  options.readPreference = resolveReadPreference(options, { db: coll.s.db, collection: coll });
+  options.readPreference = legacyResolveReadPreference(options, {
```

Comment:
why do we need `legacyResolveReadPreference` again?

Reply:
I used `legacyResolveReadPreference` for dead code paths. This function should never be called because `collection.distinct` uses `lib/operations/distinct`. I can switch this to `resolveReadPreference` if you'd prefer, though.

Reply:
if this is a dead code path I would rather delete the whole method than create a new function to support dead code